src/index.ts

```diff
@@ -854,6 +869,98 @@ export abstract class PinejsClientCore<PinejsClient> {
 		}
 	}
 
+	private waitAsync(delayMs: number): Promise<void> {
+		return new Promise<void>((resolve) => {
+			setTimeout(resolve, delayMs);
+		});
+	}
+
+	private canRetryDefaultHandler(err: any) {
+		const code = err?.statusCode ?? 0;
+		return code === 0 || code === 429 || (code >= 500 && code < 600);
```

Comment:
Do you expect `code === 0` to happen outside of the default? If not then I'd prefer
```suggestion
		const code = err?.statusCode;
		return code == null || code === 429 || (code >= 500 && code < 600);
```